java/dagger/internal/codegen/BindingGraph.java

```diff
@@ -1015,10 +1015,11 @@ void resolve(BindingKey bindingKey) {
         private final Set<Object> cycleChecker = new HashSet<>();
 
         /**
-         * Returns {@code true} if any of the bindings resolved for {@code bindingKey} are
-         * multibindings with contributions declared within this component's modules or optional
-         * bindings with present values declared within this component's modules, or if any of its
-         * unscoped dependencies depend on such bindings.
+         * Returns {@code true} if any of the bindings resolved for {@code bindingKey} are provision
+         * bindings that conflict with previously resolved bindings or multibindings with
```

Comment:
Ultimately, we're returning true if the binding and its transitive dependencies, as resolved in this component, will be different than what was resolved in the parent. Can we just say that?

I still wonder whether this attempt at optimizing buys us anything. Why not just resolve the binding again here, and then compare that subgraph to the one generated in the parent component? I mean, if you could tell that they weren't going to be different without actually checking all the transitive dependencies, then it might be worth it because you could reduce build times. But aren't we basically checking it here, only to throw all that away and redo it if it's different?

But that is not for now in this PR. For now, this is fine.